runners/portability/java/build.gradle

```diff
@@ -82,8 +87,15 @@ void execBackgroundInVirtualenv(String... args) {
   proc.waitFor();
 }
 
-task startLocalJobService {
+task installBeamPythonInVirtualenv {
   dependsOn setupVirtualenv
+  doLast {
+    execInVirtualenv "pip", "install", "-e", "."
```

Comment:
I did not understand previously that the Beam SDK and requirements were not installed into the virtualenv. Again, masked by the fact that I only ran it with `-PlocalJobServicePortFile` hence did not test the virtualenv setup properly.

Reply:
We had issues in the past with installing SDK in developer mode in tests. Not sure if this still relevant. If you want to install from tarball, you can consider using https://github.com/apache/beam/blob/943a948e5a6dd7ec81760359c198c3fb2b3c3dcd/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L1999

(nameing is for that target is not very mnemonic..)

Reply:
I did not use that because I read the name and it seemed tied to GCP.

Reply:
So you mean more of an approach like this?

```
       dependsOn ':sdks:python:sdist'
        doLast {
          def distTarBall = "${pythonRootDir}/build/apache-beam.tar.gz"
          project.exec {
            executable 'sh'
            args '-c', ". ${project.ext.envdir}/bin/activate && pip install --retries 10 ${distTarBall}"
          }
        }
      }
```

I am not familiar with the install variants. For ValidatesRunner we should not need any cloud-specific things. I do not know what `test` changes.

Reply:
it currently installs all `extra` dependencies (test deps + deps for all of the supported clouds)

Reply:
Race condition between comment updates, did not see your prior comments.

[test] installs additional dependencies mentioned in https://github.com/apache/beam/blob/e6b3cf1da5f17a3a15f2a48986a1b84266f3a64c/sdks/python/setup.py#L176
Same applies for other 'extra' dependencies.

Your proposal sounds good to me. I would  move :
installBeamPythonInVirtualenv into BeamModulePlugin.groovy -> installBeamPython, but feel free to keep here if you feel strongly about it.

In a future change we can rename installGcpTest to installBeamPythonWithAllExtraDeps